flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/committer/IcebergCommitter.java

```diff
@@ -0,0 +1,434 @@
+      new CommitRequest<IcebergCommittable>() {
+        @Override
+        public IcebergCommittable getCommittable() {
+          return new IcebergCommittable(EMPTY_MANIFEST_DATA, "jobId", "operatorId", -1L);
+        }
+
+        @Override
+        public int getNumberOfRetries() {
+          return 0;
+        }
+
+        @Override
+        public void signalFailedWithKnownReason(Throwable t) {}
+
+        @Override
+        public void signalFailedWithUnknownReason(Throwable t) {}
+
+        @Override
+        public void retryLater() {}
+
+        @Override
+        public void updateAndRetryLater(IcebergCommittable committable) {}
+
+        @Override
+        public void signalAlreadyCommitted() {}
+      };
+
+  @VisibleForTesting
+  static final String MAX_CONTINUOUS_EMPTY_COMMITS = "flink.max-continuous-empty-commits";
+
+  public static final long INITIAL_CHECKPOINT_ID = -1L;
+  public static final String FLINK_JOB_ID = "flink.job-id";
+  public static final String OPERATOR_ID = "flink.operator-id";
+  private final String branch;
+  private final Map<String, String> snapshotProperties;
+  private final boolean replacePartitions;
+  private final int workerPoolSize;
+  private final String prefix;
+  private transient IcebergFilesCommitterMetrics committerMetrics;
+  private transient Table table;
+  private transient int maxContinuousEmptyCommits;
+  private transient ExecutorService workerPool;
+  private transient int continuousEmptyCheckpoints = 0;
+
+  public IcebergCommitter(
+      TableLoader tableLoader,
+      String branch,
+      Map<String, String> snapshotProperties,
+      boolean replacePartitions,
+      int workerPoolSize,
+      String prefix,
+      IcebergFilesCommitterMetrics committerMetrics) {
+    this.branch = branch;
+    this.snapshotProperties = snapshotProperties;
+    this.replacePartitions = replacePartitions;
+    this.workerPoolSize = workerPoolSize;
+    this.prefix = prefix;
+    this.committerMetrics = committerMetrics;
+
+    if (!tableLoader.isOpen()) {
+      tableLoader.open();
+    }
+
+    this.table = tableLoader.loadTable();
+    this.maxContinuousEmptyCommits =
+        PropertyUtil.propertyAsInt(table.properties(), MAX_CONTINUOUS_EMPTY_COMMITS, 10);
+    Preconditions.checkArgument(
+        maxContinuousEmptyCommits > 0, MAX_CONTINUOUS_EMPTY_COMMITS + " must be positive");
+    this.workerPool =
+        ThreadPools.newWorkerPool("iceberg-worker-pool-" + table + "-" + prefix, workerPoolSize);
+    this.continuousEmptyCheckpoints = 0;
+  }
+
+  @Override
+  public void commit(Collection<CommitRequest<IcebergCommittable>> commitRequests)
+      throws IOException, InterruptedException {
+    NavigableMap<Long, byte[]> manifestMap = Maps.newTreeMap();
+    NavigableMap<Long, CommitRequest<IcebergCommittable>> commitRequestMap = Maps.newTreeMap();
+
+    if (commitRequests.isEmpty()) {
+      commitRequests.add(EMPTY_COMMIT_REQUEST);
```

Comment:
adding the empty commit will make the code to keep flowing and reaching  `commitPendingResult` method
so that this code is executed and `continuousEmptyCheckpoints` is updated
```java
 continuousEmptyCheckpoints = totalFiles == 0 ? continuousEmptyCheckpoints + 1 : 0;
```